polyfills/settings/settings.js

```diff
@@ -136,55 +118,72 @@
   var _observers = {};
 
   // And this is something else that might be reusable...
-  function Observer(setting, callback) {
+  function Observer(reqId, extraData) {
     this._id = null;
+    _observers[extraData.settingName][extraData.callback] = this;
     this.serialize = () => {
       if (!this._id) {
-        this._id = ++_currentRequestId;
+        this._id = reqId;
       }
       return {
         id: this._id,
-        data: {
-          operation: 'addObserver',
-          settingName: setting
-        },
-        processAnswer: answer => callback(answer.data)
+        data: extraData,
+        processAnswer: answer => extraData.callback(answer.event)
       };
     };
   }
 
-  function ObserverRemoval(observer) {
+  function ObserverRemoval(reqId, extraData) {
     this.serialize = () => {
       return {
-        id: observer._id,
-        data: {
-          operation: 'removeObserver',
```

Comment:
Same as before. In fact, since those two objects are quite similar, you can just do a:
function ObserverOp(op, reqId, extraData) {...
}
var ObserverRemoval = ObserverOp.bind/undefined, 'removeObserver');
var Observer = ObserverOp.bind/undefined, 'addObserver');